io-engine/src/lvm/lv_replica.rs

```diff
@@ -0,0 +1,642 @@
+                },
+                |s| s,
+            );
+            return Err(Error::FailedExec {
+                err: msg,
+            });
+        }
+
+        info!("lvm volume {} deleted", self.lv_name);
+
+        Ok(())
+    }
+
+    /// Resize the logical volume to the given size.
+    pub(crate) async fn resize(&self) -> Result<(), Error> {
+        todo!()
+    }
+
+    async fn set_share_protocol(
+        &mut self,
+        protocol: Protocol,
+    ) -> Result<(), Error> {
+        self.set_property(".lv.share", protocol.as_ref()).await?;
+        self.share = protocol;
+        Ok(())
+    }
+    async fn set_allowed_hosts(
+        &mut self,
+        hosts: &[String],
+    ) -> Result<(), Error> {
+        let hosts = hosts.join(",");
+        self.set_property(".lv.allowed_hosts", &hosts).await?;
+        Ok(())
+    }
+
+    async fn set_property(&self, key: &str, value: &str) -> Result<(), Error> {
+        let mut args = vec![format!("--addtag={LVM_TAG}{key}={value}")];
+        if let Some((tag, old_value)) = self.property_tag(key) {
+            if old_value == value {
+                // well, no point doing anything right?
+                return Ok(());
+            }
+            // remove the old tag
+            args.push(format!("--deltag={tag}"));
+        }
+
+        let output = Command::new(LVCHANGE_COMMAND)
+            .args(args)
+            .arg(&self.path)
+            .output()
+            .await?;
+
+        if !output.status.success() {
+            let msg = String::from_utf8(output.stderr).map_or_else(
+                |e: std::string::FromUtf8Error| {
+                    format!(
+                        "failed to parse stderr for lvchange modifytag: {}",
+                        e.to_string()
+                    )
+                },
+                |s| s,
+            );
+            return Err(Error::FailedExec {
+                err: msg,
+            });
+        }
+        Ok(())
+    }
+
+    async fn bdev_share_props(
+        bdev: &mut UntypedBdev,
+        protocol: Protocol,
+        ptpl: impl PtplFileOps,
+        allowed_hosts: Vec<String>,
+    ) -> Result<(), Error> {
+        match protocol {
+            Protocol::Nvmf => {
+                let props = ShareProps::new()
+                    .with_allowed_hosts(allowed_hosts)
+                    .with_ptpl(ptpl.create().map_err(|source| {
+                        Error::BdevShare {
+                            source: crate::core::CoreError::Ptpl {
+                                reason: source.to_string(),
+                            },
+                        }
+                    })?);
+                Self::bdev_nvmf_props(bdev, Some(props)).await?;
+            }
+            Protocol::Off => {
+                Self::bdev_none_props(bdev).await?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Import the Logical Volume by loading it via an SPDK bdev using either
+    /// AIO or Uring.
```

Comment:
Maybe specify only AIO for the time being?

Reply:
Maybe we can specify this somehow?